### DISCUSSION TITLE
Perplexity AI

### DISCUSSION THREAD

**Discussion by wudnerk** [ITSpecialist111/ai_automation_suggester#34] [Ideas]
I have some credit for Perplexity AI, how to use it?

**Comment by ITSpecialist111** [comment]
Hey, I haven't implemented any APIs for Perplexity at the moment but will see if we can add them into a near future release. The next priority is to enhance the suggestion substrate, but I will move this request across to the ideas sections in the discussions tab.

**Comment by wudnerk** [comment]
Hey, any updates on this?

**Comment by zobe123** [comment]
Hey, any updates on this?

**Reply by ITSpecialist111** [reply]
I'm afraid I haven't got around to building it into the code yet, however, I had done some checks we look at the similarities of the OpenAI get/post responses and Perplexity's - I also don't have a paid key to try (as I have a lot with other providers atm). I'd like you to try something and feedback if possible: 

Try Perplexity via the Custom OpenAI Option
Open the Integration’s Options

Go to Settings > Devices & Services in Home Assistant.
Find your AI Automation Suggester entry, click Configure or Options.
Select Custom OpenAI

Ensure your provider is set to “Custom OpenAI.”
Fill Out the Fields

custom_openai_endpoint:
Enter https://api.perplexity.ai/v1/chat/completions
custom_openai_api_key:
Paste in your Perplexity API key. If Perplexity requires a Bearer token, add "Bearer " before your key (e.g., "Bearer YOUR_KEY_HERE").
custom_openai_model:
Type in a valid Perplexity model name, for example "pplx-3b".
max_tokens:
Set an appropriate limit, such as 1024 (or your preferred token cap).
Save & Test

Click Submit to save your configuration.
Return to Home Assistant and try generating automation suggestions (via scanning new entities, a custom prompt, or any method you normally use).
Monitor the logs for any errors and see if suggestions come back successfully.
Provide Feedback

If everything works, you’ll see AI suggestions generated through Perplexity!
If you encounter errors (like a KeyError or unexpected JSON), Perplexity’s API might not match exactly what our integration expects.
 
 Let me know what you see in the logs so we can investigate further.

**Reply by zobe123** [reply]
Hello,

I have configured the Custom OpenAI option as follows:

custom_openai_endpoint: https://api.perplexity.ai/v1/chat/completions
custom_openai_api_key: "Bearer API_KEY"
custom_openai_model: sonar
max_tokens: 512
However, I’m receiving the following log messages:

2025-03-08 19:17:02.894 DEBUG (MainThread) [custom_components.ai_automation_suggester.config_flow] Validating Custom OpenAI endpoint https://api.perplexity.ai/v1/chat/completions
2025-03-08 19:17:03.259 ERROR (MainThread) [custom_components.ai_automation_suggester.config_flow] Custom OpenAI validation error response: 
2025-03-08 19:17:26.065 DEBUG (MainThread) [custom_components.ai_automation_suggester.config_flow] Validating Custom OpenAI endpoint https://api.perplexity.ai/v1/chat/completions
2025-03-08 19:17:26.257 ERROR (MainThread) [custom_components.ai_automation_suggester.config_flow] Custom OpenAI validation error response:
Could you please help me understand what might be going wrong?

Thanks in advance!
